tests/ieee802154_hal/main.c

```diff
@@ -56,17 +76,34 @@ static uint8_t seq;
 static void _print_packet(size_t size, uint8_t lqi, int16_t rssi)
 {
     if (buffer[0] & IEEE802154_FCF_TYPE_ACK && ((seq-1) == buffer[2])) {
-        printf("Received valid ACK with sqn %i\n", buffer[2]);
+            DEBUG_PRINT("Received valid ACK with sqn %i\n", buffer[2]);
+        received_acks++;
     }
     else {
-        puts("Packet received:");
-        for (unsigned i=0;i<size;i++) {
-            printf("%02x ", buffer[i]);
+        if (ENABLE_DEBUG) {
+            size_last_packet = size;
+            puts("Packet received:");
+            for (unsigned i=0;i<size;i++) {
+                printf("%02x ", buffer[i]);
+            }
+        }
```

Comment:
```suggestion
        size_last_packet = size;
        DEBUG("Packet received:\n");
        for (unsigned i=0;i<size;i++) {
            DEBUG("%02x ", buffer[i]);
        }
```

It should be enough to use DEBUG instead of puts

Reply:
After looking this with more details, the only case we would like to show the results is when the test is not running (`send_reply == false`). Otherwise the test fails.